spec/date_range_spec.rb

```diff
@@ -0,0 +1,105 @@
+require_relative 'spec_helper'
+
+describe 'DateRange class' do
+  let (:today) { Date.today }
+  let (:today_plus_4) { today + 4}
+  let (:today_minus_2) {today - 2}
+  let (:range) { Hotel::DateRange.new(today, today_plus_4) }
+
+  let (:same_dates) { Hotel::DateRange.new(today, today_plus_4) }
+  let (:overlap_front) { Hotel::DateRange.new(today_minus_2, today + 1) }
+  let (:overlap_back) { Hotel::DateRange.new(today + 1, today_plus_4 + 1) }
+  let (:contained) { Hotel::DateRange.new(today + 1, today_plus_4 - 1) }
+  let (:containing) { Hotel::DateRange.new(today_minus_2, today_plus_4 + 1) }
+
+  let (:before_dates) { Hotel::DateRange.new(today_minus_2 - 1, today_minus_2) }
+  let (:after_dates) { Hotel::DateRange.new(today_plus_4 + 1, today_plus_4 + 2) }
+  let (:ends_on_checkin) { Hotel::DateRange.new(today_minus_2, today) }
+  let (:starts_on_checkout) { Hotel::DateRange.new(today_plus_4, today_plus_4 + 1) }
+
+  describe 'DateRange instantiation' do
+
+    it 'creates an instance of DateRange' do
+      expect(Hotel::DateRange.new(today, today_plus_4)).must_be_instance_of Hotel::DateRange
+
+      expect(Hotel::DateRange.new(today, today_plus_4)).must_respond_to :check_in
```

Comment:
I ended up changing `today_minus_2` to `before_today` and `today_plus_4` to `week_later`. The inline calculations made sense too. I was just wondering if renaming the variables with some temporal reference versus what I had before (like writing out an equation?) makes it any better. Otherwise, I'm happy to change to inline.